### PR TITLE
Added the ability for CMake to generate Debian packages for Souffle

### PR DESCRIPTION
**Pull request by phao5814**
Modified the CMakeLists.txt file so that DEB and RPM packages are automatically generated by CMake. Also created a new `create-packages.yml` GitHub Actions file that runs whenever there is a "release" action, which triggers a suite of private actions. Each private Github Action generates a package for a target operating system and architecture, and then pushes this package to packagecloud.io.

### CODE REVIEW

File: .github/actions/create-package/ubuntu/18.04/32bit/action.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+name: Package for Ubuntu 21.04 64bit
+description: Package for Ubuntu 21.04 64bit
```
Comment (XiaowenHu96) [inline .github/actions/create-package/ubuntu/18.04/32bit/action.yml:2]: Should be 18.04

File: .github/actions/create-package/entrypoint.sh [hunk 2]
```diff
@@ -0,0 +1,21 @@
+#!/bin/sh
+
+PACKAGE_CLOUD_API_KEY="$1"
+
+# Run the build command
+case "$ARCHITECTURE" in
```
Comment (XiaowenHu96) [inline .github/actions/create-package/entrypoint.sh:6]: The name architecture is a little bit confusing, maybe try `DOMAIN_SIZE`?

File: .github/actions/create-package/ubuntu/18.04/32bit/action.yml (hunk 1 above)
Comment (phao5814) [reply .github/actions/create-package/ubuntu/18.04/32bit/action.yml]: Fixed problematic naming.

File: .github/actions/create-package/entrypoint.sh (hunk 2 above)
Comment (phao5814) [reply .github/actions/create-package/entrypoint.sh]: Changed `ARCHITECTURE` to `DOMAIN_SIZE`.

**Comment by b-scholz** [comment]
There is a conflict. Could you resolve it?

**Review by b-scholz — APPROVED** [review]

**Comment by b-scholz** [comment]
Thanks for the contribution!

**Comment by phao5814** [comment]
You're welcome Bernhard! 🙂

**Comment by b-scholz** [comment]
This is related to #1856

**Pull request merged**